joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForPackageConstructorCreator.scala

```diff
@@ -0,0 +1,51 @@
+package io.joern.gosrc2cpg.astcreation
+
+import io.joern.x2cpg.{Ast, AstCreatorBase, ValidationMode, Defines as XDefines}
+import io.shiftleft.codepropertygraph.generated.{EvaluationStrategies, NodeTypes}
+import io.shiftleft.codepropertygraph.generated.nodes.{NewBlock, NewMethod, NewMethodReturn}
+import org.apache.commons.lang.StringUtils
+import overflowdb.BatchedUpdate.DiffGraphBuilder
+
+import scala.collection.immutable.Set
+
+class AstForPackageConstructorCreator(val pacakgePath: String, statements: Set[(Ast, String)])(implicit
+  withSchemaValidation: ValidationMode
+) extends AstCreatorBase(pacakgePath) {
+
+  override def createAst(): DiffGraphBuilder = {
+    val name = StringUtils.normalizeSpace(s"$pacakgePath${XDefines.StaticInitMethodName}")
+    val fakeGlobalMethod =
+      NewMethod()
```

Comment:
@DavidBakerEffendi I had to create direct nodes instead of using x2cpg APIs that's because I didn't had a corresponding Node object from which the respective line and column number information is set from the API.

Reply:
The `node` argument in `AstNodeBuilder` is largely null-safe so the lineNumber-style variables will just be left as their default values. You would however need to update the `code` property afterwards, but this is fine.

Reply:
But I suppose the intention is that they are meant to be `0`, in which case then yeah disregard the last comment.